src/libs/sfdk/dockercontainer.cpp

```diff
@@ -0,0 +1,341 @@
+{
+    Q_Q(const DockerContainer);
+    Q_ASSERT(context);
+    Q_ASSERT(functor);
+
+    DockerManager::fetchRegisteredVirtualMachines(q,
+        [=](const QStringList &registeredVms, bool ok) {
+            if (!ok) {
+                VirtualMachinePrivate::BasicState state;
+                functor(state, false);
+            } else {
+                if (!registeredVms.contains(q->name())) {
+                    VirtualMachinePrivate::BasicState state;
+                    functor(state, true);
+                }
+                QStringList arguments;
+                arguments.append(QLatin1String(CONTAINER_LIST));
+                arguments.append(QLatin1String(FILTER_PREFIX)+q->name());
+                arguments.append(QLatin1String(FORMAT));
+                auto runner = std::make_unique<ProcessRunner>(DockerManager::dockerPath(), arguments);
+                QObject::connect(runner->process(), QOverload<int, QProcess::ExitStatus>::of(&QProcess::finished), context,
+                        [=, runner = runner.get()](int exitCode, QProcess::ExitStatus exitStatus) {
+                    VirtualMachinePrivate::BasicState state = VirtualMachinePrivate::Existing;
+                    if (exitStatus != QProcess::NormalExit) {
+                        functor(state, false);
+                        return;
+                    }
+                    if (exitCode != 0) {
+                        functor(state, false);
+                        return;
+                    }
+
+                    QString runningContainers = QString::fromLocal8Bit(runner->process()->readAllStandardOutput());
+                    bool isRunning = runningContainers.contains(q->name());
+                    if (isRunning) {
+                        state |= (VirtualMachinePrivate::Running | VirtualMachinePrivate::Headless);
+                        if (q->m_containerId.isEmpty())
+                            q->m_containerId = runningContainers.split(' ')[1].trimmed();
+                    }
+                    functor(state, true);
+                });
+                SdkPrivate::commandQueue()->enqueue(std::move(runner));
+
+            }
+        });
+}
+
+void DockerContainerPrivate::setVideoMode(const QSize &size, int depth, const QString &deviceModelName,
+                                          Qt::Orientation orientation, int scaleDownFactor, const QObject *context,
+                                          const Functor<bool> &functor)
+{
+    Q_UNUSED(size)
+    Q_UNUSED(depth)
+    Q_UNUSED(deviceModelName)
+    Q_UNUSED(orientation)
+    Q_UNUSED(scaleDownFactor)
+    Q_UNUSED(context)
+    Q_UNUSED(functor)
+    Q_ASSERT(false);
+}
+
+void DockerContainerPrivate::doSetMemorySizeMb(int memorySizeMb, const QObject *context, const Functor<bool> &functor)
+{
+    Q_UNUSED(memorySizeMb)
+    Q_UNUSED(context)
+    Q_UNUSED(functor)
```

Comment:
assert false? similar in other unimplemented functions

Reply:
fixed